test/unittests/parser_test.cpp

```diff
@@ -465,3 +465,63 @@ TEST(parser, milestone1)
                                      "02000000"
                                      "00000000"));
 }
+
+TEST(parser, instr_loop)
+{
+    const auto loop_without_imm = from_hex("030b");
+    EXPECT_THROW(parse_expr(loop_without_imm.data()), parser_error);
+
+    const auto loop_void_empty = from_hex("03400b0b");
+    const auto [code1, pos1] = parse_expr(loop_void_empty.data());
+    EXPECT_EQ(code1.instructions, (std::vector{Instr::loop, Instr::end, Instr::end}));
+    EXPECT_EQ(code1.immediates.size(), 0);
+
+    const auto loop_i32_empty = from_hex("037f0b0b");
+    EXPECT_THROW(parse_expr(loop_i32_empty.data()), parser_error);  // loop arity != 0.
+}
+
+TEST(parser, instr_block)
+{
+    const auto wrong_type = from_hex("0200");
+    EXPECT_THROW(parse_expr(wrong_type.data()), parser_error);
+
+    const auto empty = from_hex("010102400b0b");
+    const auto [code1, pos1] = parse_expr(empty.data());
+    EXPECT_EQ(code1.instructions,
+        (std::vector{Instr::nop, Instr::nop, Instr::block, Instr::end, Instr::end}));
+    EXPECT_EQ(hex(code1.immediates), "000400000009000000");
```

Comment:
would prefer this formatting
```suggestion
    EXPECT_EQ(hex(code1.immediates), "00" + "04000000" + "09000000");
```